e2e_test/e2e_test.go

```diff
@@ -369,6 +370,125 @@ func TestStartStopValidators(t *testing.T) {
 
 }
 
+func runStopNetworkAtL2BlockTest(ctx context.Context, t *testing.T, network test.Network, l2Block *big.Int) {
+	err := network.AwaitBlock(ctx, l2Block.Uint64()-1)
+	require.NoError(t, err)
+
+	shortCtx, cancel := context.WithTimeout(ctx, time.Second*2)
+	defer cancel()
+
+	// fail if any node adds a new block >= the migration block
+	var wg sync.WaitGroup
+	for _, n := range network {
+		wg.Add(1)
+		go func(n *test.Node) {
+			defer wg.Done()
+			err = n.Tracker.AwaitBlock(shortCtx, l2Block.Uint64())
+			require.EqualError(t, err, context.DeadlineExceeded.Error())
```

Comment:
`T.FailNow` which is ultimately called by all `require.XXXX` only functions correctly when called from the main test goroutine. Essentially any assertions you do in tests need to be made in the main goroutine, so this is not going to work correctly. Instead you can use the `errgroup` package in place of the waitgroup and do your error assertion at the wait point.

Reply:
Thanks! I was not aware of that. I looked into using errgroup for this but then remembered that we actually want to wait for all the nodes to time out here, and errgroup will only wait until the first error is returned. I ended up just using a channel to pass the error back to the main go routine https://github.com/celo-org/celo-blockchain/pull/2322/commits/c7bea3217e78c76eb96958ccfd27592f99afaba4